vm/src/obj/objset.rs

```diff
@@ -57,6 +57,336 @@ impl PyValue for PyFrozenSet {
     }
 }
 
+trait SetProtocol
+where
+    Self: Sized,
+{
+    fn get_elements(&self) -> HashMap<u64, PyObjectRef>;
```

Comment:
we definitely don't want to copy the entire map just to eg check its size.

Reply:
This is the same as the current implementation. I would prefer not changing the functionality of the code in this PR to make it easier to review. This is a big diff and I think improving in a different PR will be better.

I did add `__iter__` to `frozenset` but used the same code as already existed in `set`.